sdk/v2/core/projects.go

```diff
@@ -172,17 +173,19 @@ type ProjectsClient interface {
 	// Delete deletes a single Project specified by its identifier.
 	Delete(context.Context, string) error
 
-	// Roles returns a specialized client for Project Role management.
-	Roles() ProjectRolesClient
+	// RoleAssignments returns a specialized client for managing project-level
+	// RoleAssignments.
+	RoleAssignments() authx.RoleAssignmentsClient
```

Comment:
Project-level roles are "special," but minimally so. The scope of a project-level role == the ID of the project. On the server side, when you assign a principal to a project-level role (as opposed to a system-level role), we need to make sure the project exists! That's a concern that doesn't exist for system-level roles. Owing to this, there are (or will be, actually) separate endpoints on the API server for managing system-level and project-level role assignments. They'll actually share most of their code, but two separate endpoints == two separate clients.